File: support_tools/swigwin-3.0.12/Examples/test-suite/octave/director_string_runme.m
# do not dump Octave core
if exist("crash_dumps_octave_core", "builtin")
  crash_dumps_octave_core(0);
endif

director_string


function out=get_first(self)
  out = strcat(self.A.get_first()," world!");
end
function process_text(self,string)
  self.A.process_text(string);
  self.smem = "hello";
end
B=@(string) subclass(A(string),'get_first',@get_first,'process_text',@process_text);


b = B("hello");

b.get(0);
if (!strcmp(b.get_first(),"hello world!"))
  error(b.get_first())
endif

b.call_process_func();

if (!strcmp(b.smem,"hello"))
  error(b.smem)
endif

  

File: support_tools/swigwin-3.0.12/Examples/test-suite/go/director_string_runme.go
package main

import . "./director_string"

type B struct { // From A
	abi  A
	smem string
}

func NewB(s string) A {
	p := &B{nil, ""}
	ret := NewDirectorA(p, s)
	p.abi = ret
	return ret
}

func (p *B) Get_first() string {
	return DirectorAGet_first(p.abi) + " world!"
}

func (p *B) Process_text(s string) {
	DirectorAProcess_text(p.abi, s)
	p.smem = "hello"
}

func main() {
	b := NewB("hello")

	b.Get(0)
	if b.Get_first() != "hello world!" {
		panic(b.Get_first())
	}

	b.Call_process_func()

	if b.DirectorInterface().(*B).smem != "hello" {
		panic(b.DirectorInterface().(*B).smem)
	}
}
